base/strings/string.jl

```diff
@@ -231,20 +231,21 @@ isvalid(s::String, i::Integer) =
 
 function getindex(s::String, r::UnitRange{Int})
     isempty(r) && return ""
-    i, j = first(r), last(r)
     l = sizeof(s)
-    if i < 1 || i > l
```

Comment:
Just a styling issue, but I don't see the point of changing these from `if` to `&&`, especially since you add a new `if` block for `UnicodeError`.